### PR TITLE
Upgrade replication client to v11

### PR DESCRIPTION
**Pull request by fjammes**
Bump replication service API to v11
Add support for replication service parameter: 'http_host_name'

### CODE REVIEW

File: dev_local.sh [hunk 1]
```diff
@@ -35,24 +35,30 @@ if [ $# -ne 0 ] ; then
     exit 2
 fi
 
-echo -n "Check telepresence version "
-if [ ! telepresence version | grep -q "Client: v2" ]
+TELEPRESENCE_VERSION="v2"
+if telepresence version | grep "Client: $TELEPRESENCE_VERSION" > /dev/null
```
Comment (andy-slac) [inline dev_local.sh]: This will also match `Client: v2000-01-01`, maybe use more exact match?

File: example/empty_queue.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+# Remove ingest queue entries for a given database 
+
+# Database names:
+# - cosmoDC2_v1_1_4_image_overlap
+# - dc2_run2_1i_dr1b
+# - dc2_errors
+# - dc2_object_run22i_dr6_wfd_v2_00
+
+# Default password
+# PASSWORD=CHANGEME
```
Comment (andy-slac) [inline example/empty_queue.sh]: Do you need all of these comments here (especially password)?

File: rootfs/ingest/python/qserv/http.py [hunk 3]
```diff
@@ -31,6 +31,7 @@
 #  Imports of standard modules --
 # -------------------------------
 import getpass
+from http.client import RemoteDisconnected
```
Comment (andy-slac) [inline rootfs/ingest/python/qserv/http.py]: This is not used in the code?

File: rootfs/ingest/python/qserv/jsonparser.py [hunk 4]
```diff
@@ -161,9 +163,33 @@ def get_chunk_location(responseJson: dict) -> Tuple[str, int]:
...
+
+    Parameters
+    ----------
+    fqdns: `str`
+        comma-separated list of fqdns
+    port: `int`
+        url port to reach
+
+    Returns:
+        str: first reachable host fqdn, empty string if not fqdn is reachable
```
Comment (andy-slac) [inline rootfs/ingest/python/qserv/jsonparser.py]: This should read:

    Returns
    -------
    fdqn : `str`
        First reachable host fqdn, empty string if not fqdn is reachable.

File: rootfs/ingest/python/qserv/jsonparser.py [hunk 5]
```diff
@@ -184,9 +210,10 @@ def get_regular_table_locations(responseJson: dict) -> List[Tuple[str, int]]:
     """
     locations = []
     for entry in responseJson["locations"]:
-        host = entry["http_host"]
+        fqdns = entry["http_host_name"]
         port = entry["http_port"]
-        locations.append((host, port))
+        fqdn = get_fqdn(fqdns, port)
```
Comment (andy-slac) [inline rootfs/ingest/python/qserv/jsonparser.py:217]: Do you want to check for empty `fqdn` here?

File: rootfs/ingest/python/qserv/replicationclient.py [hunk 6]
```diff
@@ -46,7 +46,7 @@
 # Local non-exported definitions --
 # ---------------------------------
 _LOG = logging.getLogger(__name__)
-_VERSION = 9
+_VERSION = 11
```
Comment (andy-slac) [inline rootfs/ingest/python/qserv/replicationclient.py]: Is there version 10 already on some other branch?

File: rootfs/ingest/python/qserv/test_jsonparser.py [hunk 7]
```diff
@@ -63,6 +63,14 @@ def test_contribution_monitor() -> None:
     assert contrib_monitor.retry_allowed is False
 
 
+def test_get_fqdn() -> None:
+    fqdn = jsonparser.get_fqdn("k8s-school.fr", 80)
+    assert fqdn == "k8s-school.fr"
+
+    fqdn = jsonparser.get_fqdn("does-not-exists,k8s-school.fr", 80)
+    assert fqdn == "k8s-school.fr"
```
Comment (andy-slac) [inline rootfs/ingest/python/qserv/test_jsonparser.py]: Assuming that that domain is reachable from all locations sounds very wrong.

**Review by andy-slac — APPROVED** [review]
Looks OK, few comments.

File: dev_local.sh (hunk 1 above)
Comment (fjammes) [reply dev_local.sh]: :+1:

File: example/empty_queue.sh (hunk 2 above)
Comment (fjammes) [reply example/empty_queue.sh]: :+1:

File: rootfs/ingest/python/qserv/http.py (hunk 3 above)
Comment (fjammes) [reply rootfs/ingest/python/qserv/http.py]: :+1:

File: rootfs/ingest/python/qserv/jsonparser.py (hunk 4 above)
Comment (fjammes) [reply rootfs/ingest/python/qserv/jsonparser.py]: :+1:

File: rootfs/ingest/python/qserv/jsonparser.py (hunk 5 above)
Comment (fjammes) [reply rootfs/ingest/python/qserv/jsonparser.py:217]: :+1:

File: rootfs/ingest/python/qserv/replicationclient.py (hunk 6 above)
Comment (fjammes) [reply rootfs/ingest/python/qserv/replicationclient.py]: Version is for replication service, and has bump from 9 to 11 while I was coding on this ticket.

Comment (fjammes) [reply rootfs/ingest/python/qserv/replicationclient.py]: I'll rename _VERSION to _REPL_SERVICE_VERSION for clarity

**Pull request merged**